ConcurrentAPI.adoc

```diff
@@ -0,0 +1,255 @@
+== Distributed Concurrent API
+
+=== Motivation
+
+Coming from an initial issue where we need to provide an implementation for vert.x cluster manager for the shared data feature,
+provide concurrent and distributed programming capabilities.
+
+* Distributed Locks
+* Distributed CountDownLatch
+
+=== Scope
+
+All these interfaces should be placed in commons, in order to be used either in Embedded and Server mode.
+
+== Infinispan Clustered Locks
+
+=== ClusteredLockManager and configuration
+
+```java
+package org.infinispan.lock.api;
+
+public class ClusteredLockManager {
+   boolean defineLock(String name, LockConfiguration configuration);
+   InfinispanLock get(String name);
+   LockConfiguration getConfiguration(String name);
+   boolean isDefined(String name);
+   CompletableFuture<Boolean> remove(String name);
+   CompletableFuture<Void>	unlock(String name);
+}
+
+
+public class LockConfiguration {
+   private final RentrancyLevel lentrancyLevel; // default None
+   private final boolean silentFailover; // default true
+   private final long maxLeaseTime; // after a time the lock is released in milles
+   private final long renewalLeaseTime; // time after a client can aquire the lock
+}
+
+public enum RentrancyLevel {
+   None, Node, Thread
```

Comment:
Sorry, I wasn't clear: `NOT_REENTRANT`, `NODE` and `INSTANCE`. Just didn't want two modes to differ by single letter only.